src/storage/state.rs

```diff
@@ -0,0 +1,151 @@
+use std::collections::HashMap;
+
+use derive_builder::Builder;
+use semver::Version;
+use serde::{Deserialize, Serialize};
+
+use crate::events::Linkdef;
+
+////////////
+// Claims //
+////////////
+
+/// A Claim that can be applied to a given lattice
+#[derive(Debug, Serialize, Deserialize, Clone, Builder, PartialEq, Eq)]
+#[builder(field(public))]
```

Comment:
Same thing here with the builder stuff on all these structs. It is just easier to derive `Default` and then let people fill out the fields they need

Reply:
The reason to use Builder here was to make partial updates possible, in particular *before* I realized that's really a good idea with the guarantees NATS KV provides.

Will remove though, in the end the Lattice state has to basically go in all in one go, unfortunately.

Reply:
All the Builder stuff was removed!